File: src/Document.py
import pickle 

class Document():
    '''
    Document Class for Word Cloud Project
    
    Daniel Klein
    Computer-Based Honors Program
    The University of Alabama
    9.12.2013
    
    This class represents a document object (such as a single Supreme Court 
    opinion). A typical object will consist of the document's text and a 
    metadata object consisting of various fields relevant to the document. 
    We will subclass the metadata class for each type of document we will use.
    '''
    
    def __init__(self, doc_metadata, doc_text, output_filename):
        '''
        doc_metadata should be a Metadata object or descendent.
        doc_text should be a string.
        '''
        self.doc_metadata = doc_metadata
        self.doc_text = doc_text
        self.output_filename = output_filename
        self.word_count = self.count_words(self.doc_text)
    
    
    def count_words(self, text):
        '''
        Utility method to perform simple wordcount.
        '''
        return len(text.split())
    
    
    def write_to_file(self):
        '''
        Uses pickle to serialize and save the Document to file.
        '''
        try:
            with open(self.output_filename, 'wb') as output_file:
                pickle.dump(self, output_file)
        except IOError: 
            print "An error occurred while pickling the document -- the output file may not be writable."
            raise IOError
        return True
    
    
    def print_doc(self):
        '''
        Display document with its metadata.
        '''
        print self
        return True
    
    
    def print_metadata(self):
        '''
        Display metadata data fields and values from doc.
        '''
        self.doc_metadata.print_metadata()
        return True
    
        
    def __str__(self):
        '''
        Converts Document object to formatted string.
        '''
        document_string = ""
        document_string += str(self.doc_metadata)
        document_string += self.doc_text
        return document_string
    
    
    
    

File: src/core/javacore/Document.java
package core.javacore;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;

/**
 * @author Daniel Klein
 * This class represents a document object (such as a single Supreme Court 
 * opinion). A typical object will consist of the document's text and a 
 * metadata object consisting of various fields relevant to the document. 
 * We will subclass the metadata class for each type of document we will use.
 */
public class Document 
{
	private Metadata docMetadata;
	private String docText;
	private String outputFilename;
	private int wordCount;
	
	public Document(Metadata docMetadata, String docText, String outputFilename)
	{
		this.docMetadata = docMetadata;
		this.docText = docText;
		this.outputFilename = outputFilename;
		this.wordCount = countWords(docText);
	}
	
	public int countWords(String text)
	{
		return text.split(" ").length;
	}
	
	public String getOutputFilename()
	{
		return this.outputFilename;
	}
	
	public String getText()
	{
		return this.docText;
	}
	
	public Metadata getMetadata()
	{
		return this.docMetadata;
	}
	
	public void serialize() throws IOException
	{
		ObjectOutputStream oos = null;
		try
		{
	        FileOutputStream fout = new FileOutputStream(this.outputFilename);
	        oos = new ObjectOutputStream(fout);
	        oos.writeObject(this);
		} catch (IOException e)
		{
			e.printStackTrace();
			System.out.println("Oops... something went wrong while serializing a Document object.");
			throw e;
		} finally
		{
			if (oos != null)
			{
				try
				{
					oos.close();
				} catch (IOException e)
				{
					e.printStackTrace();
					System.out.println("Unable to close ObjectOutputStream.");
				}
				
			}
		}
	}
	
	public String toString()
	{
		
		return "";
	}

	
}
